rskj-core/src/test/java/co/rsk/peg/StandardMultisigFederationTest.java

```diff
@@ -22,55 +22,55 @@
 import co.rsk.bitcoinj.script.ScriptBuilder;
+import co.rsk.bitcoinj.script.ScriptOpCodes;
 import org.bouncycastle.util.encoders.Hex;
 import org.ethereum.TestUtils;
 import org.ethereum.crypto.ECKey;
 import org.hamcrest.MatcherAssert;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.MockedStatic;
-import org.mockito.junit.jupiter.MockitoExtension;
 
 import java.math.BigInteger;
 import java.time.ZonedDateTime;
 import java.util.*;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.CoreMatchers.not;
 import static org.mockito.Mockito.any;
 import static org.mockito.Mockito.mockStatic;
 
-@ExtendWith(MockitoExtension.class)
-class FederationTest {
+class StandardMultisigFederationTest {
+
+    private List<BtcECKey> btcPublicKeys;
     private Federation federation;
     private List<BtcECKey> sortedPublicKeys;
     private List<ECKey> rskPubKeys;
     private List<byte[]> rskAddresses;
 
     @BeforeEach
     void createFederation() {
-        federation = new Federation(
-                FederationTestUtils.getFederationMembersFromPks(100, 200, 300, 400, 500, 600),
+        btcPublicKeys = Arrays.asList(
+            BtcECKey.fromPublicOnly(Hex.decode("023f0283519167f1603ba92b060146baa054712b938a61f35605ba08773142f4da")),
+            BtcECKey.fromPublicOnly(Hex.decode("02afc230c2d355b1a577682b07bc2646041b5d0177af0f98395a46018da699b6da")),
+            BtcECKey.fromPublicOnly(Hex.decode("031174d64db12dc2dcdc8064a53a4981fa60f4ee649a954e01bcae221fc60777a2")),
+            BtcECKey.fromPublicOnly(Hex.decode("0344a3c38cd59afcba3edcebe143e025574594b001700dec41e59409bdbd0f2a09")),
+            BtcECKey.fromPublicOnly(Hex.decode("039a060badbeb24bee49eb2063f616c0f0f0765d4ca646b20a88ce828f259fcdb9")));
+
+        federation = new StandardMultisigFederation(
+            FederationTestUtils.getFederationMembersWithKeys(btcPublicKeys),
                 ZonedDateTime.parse("2017-06-10T02:30:00Z").toInstant(),
                 0L,
                 NetworkParameters.fromID(NetworkParameters.ID_REGTEST)
         );
 
-        sortedPublicKeys = Arrays.stream(new BtcECKey[]{
-                BtcECKey.fromPrivate(BigInteger.valueOf(100)),
-                BtcECKey.fromPrivate(BigInteger.valueOf(200)),
-                BtcECKey.fromPrivate(BigInteger.valueOf(300)),
-                BtcECKey.fromPrivate(BigInteger.valueOf(400)),
-                BtcECKey.fromPrivate(BigInteger.valueOf(500)),
-                BtcECKey.fromPrivate(BigInteger.valueOf(600)),
-        }).sorted(BtcECKey.PUBKEY_COMPARATOR).collect(Collectors.toList());
-
-        rskPubKeys = Stream.of(101, 201, 301, 401, 501, 601)
-                .map(i -> ECKey.fromPrivate(BigInteger.valueOf(i)))
+        sortedPublicKeys = btcPublicKeys.stream()
```

Comment:
Do we need this? The ones above are already sorted so it's basically the same

Reply:
yesss i agree! but it was done that way and i didn't focus on refactoring the tests so much in this pr :)